WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/BackupDownloadViewModel.kt

```diff
@@ -1,15 +1,130 @@
 package org.wordpress.android.ui.jetpack.backup
 
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.annotation.StringRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
+import kotlinx.android.parcel.Parcelize
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.COMPLETE
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.DETAILS
+import org.wordpress.android.ui.jetpack.backup.BackupDownloadStep.PROGRESS
+import org.wordpress.android.util.wizard.WizardManager
+import org.wordpress.android.util.wizard.WizardNavigationTarget
+import org.wordpress.android.util.wizard.WizardState
+import org.wordpress.android.viewmodel.SingleEventObservable
+import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 
-class BackupDownloadViewModel @Inject constructor() : ViewModel() {
-    private var isStarted: Boolean = false
-    fun start() {
-        if (isStarted) {
-            return
-        }
+const val KEY_BACKUP_DOWNLOAD_ACTIVITY_ID_KEY: String = "key_backup_download_activity_id_key"
+const val KEY_BACKUP_DOWNLOAD_CURRENT_STEP = "key_backup_download_current_step"
+const val KEY_BACKUP_DOWNLOAD_COMPLETED = "key_backup_download_completed"
+const val KEY_BACKUP_DOWNLOAD_STATE = "key_backup_download_state"
+
+@Parcelize
+@SuppressLint("ParcelCreator")
+data class BackupDownloadState(
+    val siteId: Long? = null,
+    val rewindId: Long? = null
+) : WizardState, Parcelable
+
+typealias NavigationTarget = WizardNavigationTarget<BackupDownloadStep, BackupDownloadState>
+
+class BackupDownloadViewModel @Inject constructor(
+    private val wizardManager: WizardManager<BackupDownloadStep>
+) : ViewModel() {
+    private var isStarted = false
+    private var backupDownloadCompleted = false
+
+    private lateinit var backupDownloadState: BackupDownloadState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    clearOldBackupDownloadState(it)
+                    WizardNavigationTarget(it, backupDownloadState)
+                }
+        )
+    }
 
+    private val _wizardFinishedObservable = SingleLiveEvent<BackupDownloadWizardState>()
```

Comment:
Warning ⚠️ : We don't use SingleLiveEvent in new classes. We might want to consider replacing it with MutableLiveData<Event< BackupDownloadWizardState>>(). Wdyt?

Reply:
> We don't use SingleLiveEvent in new classes

That's good to know. Thanks

Reply:
In ef12c38
With the exception of `_screenTitleObservable` which changes to UiToolbarState later on down the road